src/brsTypes/components/Regex.ts

```diff
@@ -0,0 +1,153 @@
+import { BrsBoolean, BrsInvalid, BrsString, BrsValue, ValueKind } from "../BrsType";
+import { BrsComponent } from "./BrsComponent";
+import { BrsType } from "..";
+import { Callable, StdlibArgument } from "../Callable";
+import { Interpreter } from "../../interpreter";
+import { BrsArray } from "./BrsArray";
+
+export class Regex extends BrsComponent implements BrsValue {
+    readonly kind = ValueKind.Object;
+    private jsRegex : RegExp;
+
+    constructor(expression: BrsString, flags = new BrsString("")) {
+        super("roRegex");
+        this.jsRegex = new RegExp(expression.value, flags.value);
+
+        this.registerMethods([
+            this.isMatch,
+            this.match,
+            this.replace,
+            this.replaceAll,
+            this.split,
+            this.matchAll
+        ]);
+    }
+
+    toString(parent?: BrsType): string {
+        return "<Component: roRegex>";
+    }
+
+    private parseReplacementPattern(pattern: string): string {
+        return pattern.replace(/\\/g, "\$");
+    }
+
+    private isMatch = new Callable(
```

Comment:
I've been trying to keep some of the reference documentation included here to make working with these a little easier (e.g. in https://github.com/sjbarag/brs/blame/e2168603cda83b746a7101f2bb7f782b0178ce5f/src/brsTypes/components/AssociativeArray.ts#L159 where @jweeber added a simple sentence for each member function).

If you get some time, would you mind adding something small to each of the member functions here?